### PR TITLE
feat: add searchable single select component

### PR DESCRIPTION
**Pull request by Chisomchima**
Implements [DEVOPS-314](https://dhis2.atlassian.net/browse/DEVOPS-314)
---

### Problem
It seems docker hub only returns the first 500 (or 1000?) tags when retrieving the list of available tags.

### Description
This PR introduces a searchable single select component with some debouncing functionality used to query the docker tags.

---

### Screenshot:

https://github.com/user-attachments/assets/d7f76c35-1161-4d25-9227-c4c4ab0d40bd

### CODE REVIEW

File: src/components/searchable-single-select.tsx
Comment (tonsV2) [inline src/components/searchable-single-select.tsx:65]: Are we doing translations now?

File: src/components/searchable-single-select.tsx
Comment (tonsV2) [inline src/components/searchable-single-select.tsx:73]: Same here? i18n?

**Comment by tonsV2** [comment]
LGTM! But please don't merge before @Mohammer5 had a say :+1:

**Review by Mohammer5 — CHANGES_REQUESTED** [review]
I think this looks fine generally. I only found small UX things that we could optimize:

1. Right now, the search will find only those tags that match the search term 100%. Instead of checking whether the term is identical with one of the option's label, we could use the `String`'s [`.startsWith`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) or [`.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) method (I think I'd prefer `.startsWith`).
2. The loading indicator of the UI library's `SingleSelect` component is [displayed below the existing options](https://ui.dhis2.nu/demo/?path=/story/single-select--with-options-and-loading), which means that when users search for an image, they won't see the loading spinner. So instead of passing `loading` to the `SingleSelect`, I think it'd be better to pass it to the `Input`, which will display a loading spinner like [this](https://ui.dhis2.nu/demo/?path=/story/input--status-loading).

File: src/components/searchable-single-select.tsx
Comment (Chisomchima) [reply src/components/searchable-single-select.tsx:28]: this totally makes sense. 
the renaming will be seen in the next commit.

File: src/views/instances/new-dhis2/fields/image-tag-select.tsx
Comment (Mohammer5) [inline src/views/instances/new-dhis2/fields/image-tag-select.tsx:72]: This could be optimized a little more:

```suggestion
            const filteredExistingOptions = options.filter((option) => option.value.startsWith(tag))
            const existsInOptions = filteredExistingOptions.length
            setTagExists(existsInOptions)

            if (existsInOptions) {
                setOptions(filteredExistingOptions)
                return
            }
```

This way the user will get all the options that start with the input, everything else will be hidden.
I think it's safe to assume that the user doesn't care about versions he didn't look for.

**Review by Mohammer5 — APPROVED** [review]
Much better! Just one small thing, but I'll already approve here so you don't have to wait for my approval again

File: src/views/instances/new-dhis2/fields/image-tag-select.tsx
Comment (Chisomchima) [reply src/views/instances/new-dhis2/fields/image-tag-select.tsx:72]: I have to add `{ value: tag, label: tag }` to the options as the select component always checks to see if what was typed matches anything in the list.

The filteredOptions is cleanup to prevent duplicate tags from being in the option.

Writing the suggested code may not account for this, which causes errors.

Comment (Chisomchima) [reply src/views/instances/new-dhis2/fields/image-tag-select.tsx:72]: Thanks for the approval

**Pull request merged**
